src/engine.rs

```diff
@@ -121,6 +121,41 @@ impl Engine {
     pub fn get_next_span_start(&self) -> usize {
         self.engine_state.next_span_start()
     }
+
+    pub fn fetch_completions<'engine>(
+        &'engine mut self,
```

Comment:
There is no need for the lifetime sense you don't use it in the return or anywhere else